plugin/evm/atomic/sync/syncer.go

```diff
@@ -92,62 +91,75 @@ type syncer struct {
-	packer.PackLong(height)
-	packer.PackFixedBytes(zeroBytes)
-	return packer.Bytes
-}
-
-// newSyncer returns a new syncer instance that will sync the atomic trie from the network.
-func newSyncer(client syncclient.LeafClient, db *versiondb.Database, atomicTrie *atomicstate.AtomicTrie, targetRoot common.Hash, config *Config) (*syncer, error) {
-	if config.TargetHeight == 0 {
-		return nil, errInvalidTargetHeight
+// NewSyncer returns a new syncer instance that will sync the atomic trie from the network.
+func NewSyncer(client syncclient.LeafClient, db *versiondb.Database, atomicTrie *atomicstate.AtomicTrie, targetRoot common.Hash, targetHeight uint64, opts ...SyncerOption) (*Syncer, error) {
+	if targetHeight == 0 {
+		return nil, errTargetHeightRequired
 	}
 
-	// Apply defaults for unset fields.
-	cfg := config.WithUnsetDefaults()
+	cfg := config{
+		numWorkers:  defaultNumWorkers,
+		requestSize: defaultRequestSize,
+	}
+	options.ApplyTo(&cfg, opts...)
 
 	lastCommittedRoot, lastCommit := atomicTrie.LastCommitted()
 	trie, err := atomicTrie.OpenTrie(lastCommittedRoot)
 	if err != nil {
 		return nil, err
 	}
 
-	syncer := &syncer{
+	syncer := &Syncer{
 		db:           db,
 		atomicTrie:   atomicTrie,
 		trie:         trie,
 		targetRoot:   targetRoot,
-		targetHeight: cfg.TargetHeight,
+		targetHeight: targetHeight,
 		lastHeight:   lastCommit,
 	}
 
 	// Create tasks channel with capacity for the number of workers.
-	tasks := make(chan syncclient.LeafSyncTask, cfg.NumWorkers)
+	tasks := make(chan syncclient.LeafSyncTask, cfg.numWorkers)
 
 	// For atomic trie syncing, we typically want a single task since the trie is sequential.
 	// But we can create multiple tasks if needed for parallel processing of different ranges.
 	tasks <- &syncerLeafTask{syncer: syncer}
 	close(tasks)
 
 	syncer.syncer = syncclient.NewCallbackLeafSyncer(client, tasks, &syncclient.LeafSyncerConfig{
-		RequestSize: cfg.RequestSize,
-		NumWorkers:  cfg.NumWorkers,
+		RequestSize: cfg.requestSize,
+		NumWorkers:  cfg.numWorkers,
 		OnFailure:   func() {}, // No-op since we flush progress to disk at the regular commit interval.
 	})
 
 	return syncer, nil
 }
 
+// Name returns the human-readable name for this sync task.
+func (*Syncer) Name() string {
+	return "Atomic State Syncer"
+}
+
+// ID returns the stable identifier for this sync task.
+func (*Syncer) ID() string {
+	return "state_atomic_sync"
+}
+
 // Sync begins syncing the target atomic root with the configured number of worker goroutines.
-func (s *syncer) Sync(ctx context.Context) error {
+func (s *Syncer) Sync(ctx context.Context) error {
 	return s.syncer.Sync(ctx)
 }
 
+// addZeros adds [common.HashLenth] zeros to [height] and returns the result as []byte
+func addZeroes(height uint64) []byte {
+	packer := wrappers.Packer{Bytes: make([]byte, atomicstate.TrieKeyLength)}
+	packer.PackLong(height)
+	packer.PackFixedBytes(zeroBytes)
+	return packer.Bytes
+}
+
```

Comment:
```suggestion
// addZeroes returns the big-endian representation of `height`, prefixed with [common.HashLength] zeroes.
func addZeroes(height uint64) []byte {
    const n = common.HashLength
    return binary.BigEndian.AppendUint64(make([]byte, n, n+8), height)
}

```

Rationale:

1. The comment was ambiguous (add before or after?) and under-specified (endianness).
2. Using a `Packer` forces indirection, so the reader of the code has to inspect it to see what it's doing under the hood. The stdlib approach is transparent and also shorter.